features/frontend/products.feature

```diff
@@ -71,3 +84,36 @@ Feature: Products
     Scenario: Display product attributes
         Given I am on the product page for "Super T-shirt"
         Then I should see "Wool"
+
+    Scenario: Receiving exception while entering page for product with empty slug
+        Given I go to page for product with empty slug
+         Then the response status code should be 404
+
+    Scenario: Displaying product rating and reviews:
+        Given I am on the store homepage
+        And I follow "PHP T-Shirts"
+        When I click "Symfony T-Shirt"
+        Then I should see "Rating: 5"
+        And I should see 1 review on the reviews list
+
+    @javascript
+    Scenario: Adding review as anonymous user
+        Given I am on the product page for "Symfony T-Shirt"
+        When I fill in "Title" with "Very good"
+        And I fill in "Comment" with "Very good shirt."
+        And I fill in "Author" with "guest@example.com"
+        And I select the "5" radio button
+        And I press "Submit"
+        And I wait 3 seconds
```

Comment:
It's because of ajax request, that is sent during review addition. Nevertheless, we could *pack* this in a step like ``And I send review``, which inside would have all *button-pressing* and *for-response-waiting* logic ;)